---
title: "Ch2_Blueberry"
author: "MInuka"
date: "2024-07-23"
output: html_document
---

Metadata and Sampling Counts

```{r}

m_ITS1R_BC_ON_BB<-meta(PS_ITS1R_BC_ON_BB)

m_ITS1R_BB_BB<-meta(subset_samples(PS_ITS1R_BC_ON_BB,Province =="BC"))

write.csv(m_ITS1R_BB_BB, file="D:\\Grad_School\\Grad_Project\\Outputs\\Metadata\\metadata_ITS1R_BB_BB.csv")

m_ITS1R_ON_BB<-meta(subset_samples(PS_ITS1R_BC_ON_BB,Province =="ON"))

write.csv(m_ITS1R_ON_BB, file="D:\\Grad_School\\Grad_Project\\Outputs\\Metadata\\metadata_ITS1R_ON_BB.csv")





```

------------- Section 1 : Fungal Detections in BB Farms ---------------------------
Check the read depth 
Check for singletons 

```{r}

#load
load("D:\\Grad_School\\R_Projects\\BeeProject_Metagenomics\\Images\\Setup.RData")

PS_ITS1R_BC_ON_BB
meta(PS_ITS1R_BC_ON_BB)
#Read depth analysis for original samples
Read_Depth_PS_ITS1R_BC_ON_BB<-plot_read_distribution(PS_ITS1R_BC_ON_BB, groups = "Province", 
                             plot.type = "histogram")+
  theme_biome_utils()+
  scale_x_continuous(trans='log10')+
  scale_fill_manual(values=c("#111111"))+ 
  geom_vline(xintercept = 1000, colour = "black", linetype="dashed")+
  theme(legend.position="none")+
  labs(x = "", y = "Count")

Read_Depth_PS_ITS1R_BC_ON_BB

PS_ITS1R_BC_ON_BB
summarize_phyloseq(PS_ITS1R_BC_ON_BB)

#Number of singletons = 4563

#We need to remove the so we remove any taxa sums that are less than 10 reads 
PS_ITS1R_BC_ON_BB_Pruned = prune_taxa(taxa_sums(PS_ITS1R_BC_ON_BB) > 02, PS_ITS1R_BC_ON_BB)

PS_ITS1R_BC_ON_BB_Pruned

summarize_phyloseq(PS_ITS1R_BC_ON_BB_Pruned)

```


Check for Ubiquitous Fungi and remove them 
```{r Removal and Reabundance - BCC, echo=FALSE, eval=TRUE, include=FALSE, cache=TRUE}

PS_ITS1R_BC_ON_BB_Pruned

classtaxa_ITS1R_BC_ON_BB_Pruned<- get_taxadf(obj=PS_ITS1R_BC_ON_BB_Pruned, taxlevel=7)
# The 30 most abundant taxonomy will be visualized by default (parameter `topn=30`). 
TopTaxa_ITS1R_BC_ON_BB_Pruned<- ggbartax(obj=classtaxa_ITS1R_BC_ON_BB_Pruned, 
                   facetNames="Province", 
                   plotgroup=TRUE, 
                   topn=50) +
  xlab(NULL) +
  ylab("Relative abundance (%)") +
  guides(fill= guide_legend(keywidth = 0.5, keyheight = 0.5, ncol=4))

TopTaxa_ITS1R_BC_ON_BB_Pruned

#Take a closer look at the species we can remove by creating a taxa table
Taxa_Table_ITS1R_BC_ON_SF_Pruned<- as.data.frame(phyloseq::tax_table(PS_ITS1R_BC_ON_SF_Pruned))

Taxa_Table_ITS1R_BC_ON_SF_Pruned

# Removal of ubiqitous species and unknown phylums 
PS_ITS1R_BC_ON_SF_Pruned

PS_ITS1R_BC_ON_BB_Filtered <- subset_taxa(PS_ITS1R_BC_ON_BB_Pruned, Species !="Cladosporium_herbarum")
PS_ITS1R_BC_ON_BB_Filtered <- subset_taxa(PS_ITS1R_BC_ON_BB_Filtered, Species !="Cladosporium_sp")
PS_ITS1R_BC_ON_BB_Filtered <- subset_taxa(PS_ITS1R_BC_ON_BB_Filtered, Class!="Saccharomycetes")
PS_ITS1R_BC_ON_BB_Filtered <- subset_taxa(PS_ITS1R_BC_ON_BB_Filtered, Phylum!="N/A")

PS_ITS1R_BC_ON_BB_Filtered

### Did removal work ###

classtaxa_ITS1R_BC_ON_BB_Filtered <- get_taxadf(obj=PS_ITS1R_BC_ON_BB_Filtered, taxlevel=7)
# The 30 most abundant taxonomy will be visualized by default (parameter `topn=30`). 
TopTaxa_ITS1R_BC_ON_BB_Filtered<- ggbartax(obj=classtaxa_ITS1R_BC_ON_BB_Filtered, 
                   facetNames="Province", 
                   plotgroup=TRUE, 
                   topn=50) +
  xlab(NULL) +
  ylab("Relative abundance (%)") +
  guides(fill= guide_legend(keywidth = 0.5, keyheight = 0.5, ncol=4))

TopTaxa_ITS1R_BC_ON_BB_Filtered

```
Diversity analysis 
```{r Alpha , Shannon and Pileou Analysis}

PS_ITS1R_BC_ON_BB_Filtered
# Alpha diversity -> microbiotaprocess    #####

alphaobj_Blueberry<- get_alphaindex(PS_ITS1R_BC_ON_BB_Filtered)
#alphaobj2_PS_Global<-as.data.frame(alphaobj_PS_Global)
#head(alphaobj2_PS_Global)

##### violon_plots (Status) ######

Observe_status <- ggbox(alphaobj_Blueberry, 
                        geom="violin", 
                        factorNames="Province",   
                        compare = TRUE,
                        testmethod = "wilcox.test",
                        signifmap = TRUE,
                        indexNames="Observe")+ 
  theme(aspect.ratio = 0.5)+
  theme( legend.position="none")+
  theme(strip.background = element_rect(colour=NA, fill="grey"))+
  scale_fill_manual(values=c("#990038", 
                                      "#01AED9",
                                      "#999999",
                                      "#1B9E77", 
                                      "#000033", 
                                      "#FD9347"))
                                      
Shannon_status<- ggbox(alphaobj_Blueberry, 
                       geom="violin", 
                       factorNames="Province",   
                       compare = TRUE, 
                       testmethod = "wilcox.test",
                       signifmap = TRUE,
                       indexNames="Shannon")+ 
  theme(legend.position="Bottom")+
  theme(aspect.ratio = 0.5)+
  theme(strip.background = element_rect(colour=NA, fill="grey"))+
  scale_fill_manual(values=c("#990038", 
                                      "#01AED9",
                                      "#999999",
                                      "#1B9E77", 
                                      "#000033", 
                                      "#FD9347"))
                                      

Pielou_status<- ggbox(alphaobj_Blueberry, 
                      geom="violin", 
                      factorNames="Province",   
                      compare = TRUE, 
                      testmethod = "wilcox.test",
                      signifmap = TRUE,
                      indexNames="Pielou")+ 
  theme(legend.position="Bottom")+
  theme(aspect.ratio = 0.5)+
  theme(strip.background = element_rect(colour=NA, fill="grey"))+
  scale_fill_manual(values=c("#990038", 
                                      "#01AED9",
                                      "#999999",
                                      "#1B9E77", 
                                      "#000033", 
                                      "#FD9347"))
                                      
plot_grid(Observe_status+theme(legend.position="none"),
          Shannon_status+theme(legend.position="none"),
          align="vh",
          labels = c("A", "B"),
          hjust = -1,
          vjust= 3.5,
          ncol=1)


plot_grid(Observe_status+theme(legend.position="none"),
          Shannon_status+theme(legend.position="none"),
          Pielou_status+theme(legend.position="none"),
          align="vh",
          labels = c("A", "B", 'C'),
          hjust = -1,
          vjust= 3.5,
          ncol=1)

```

General Heat Map 

```{r Heat Map }

#General HeatMap 

Meco_BB <- phyloseq2meco(PS_ITS1R_BC_ON_BB_Filtered)

Heatmap_BB <- trans_abund$new(dataset = Meco_BB, 

                      taxrank = "Species", 

                      ntaxa = 15)

Heatmap_BB$plot_heatmap(facet = c("Province","Plant","Sample.type"),
                color_values = rev(RColorBrewer::brewer.pal(n = 11, 
                                                            name = "Spectral")),
                xtext_keep = FALSE, 
                withmargin = TRUE)

#BC and ON stone fruit separated metadata
PS_BC_Only_SF <- subset_samples(PS_BC_ON_SF, Province == "BC")

meta(PS_BC_Only_SF)
#82 Rows for the BC subset 

PS_ON_Only_SF <- subset_samples(PS_BC_ON_SF, Province == "ON")

meta(PS_ON_Only_SF)
#56 Rows

```

Specific Pathogen Screening

Monilinia group 
```{r BB - ON and BC -  Monilinia Group}
PS_ITS1R_Sclerotiniaceae <- subset_taxa(PS_ITS1R_BC_ON_BB_Filtered, Family =="Sclerotiniaceae")

summarize_phyloseq(PS_ITS1R_Pleosporaceae)

plot_tree(tax_glom(PS_ITS1R_Sclerotiniaceae, 
                   taxrank="Species"),
          method = "sampledodge",
          ladderize="left",
          nodelabf=nodeplotblank, 
          color="Plant", 
          label.tips="Species", 
          text.size=3, 
          base.spacing=0.01,
          justify="jagged",
          shape="Province",
          size="abundance",
          plot.margin =0.9)+
  scale_size_continuous(range = c(0.0001, 4))




meco_ITS2_BC_ON_SF_Sclerotiniaceae<- phyloseq2meco(PS_ITS2_BC_ON_SF_Sclerotiniaceae)

 
ITS2_BC_ON_SF_Sclerotiniaceae <- trans_abund$new(dataset = meco_ITS2_BC_ON_SF_Sclerotiniaceae, 

                      taxrank = "Species", 

                      ntaxa = 15)

ITS2_BC_ON_SF_Sclerotiniaceae$plot_heatmap(facet = c("Province","Sample.type"),
                color_values = rev(RColorBrewer::brewer.pal(n = 11, 
                                                            name = "Spectral")),
                xtext_keep = FALSE, 
                withmargin = TRUE)

summarize_phyloseq(PS_ITS2_BC_ON_SF_Sclerotiniaceae)

```


------------- Section 2 : Bacterial Detections in Stone Fruit Farms ---------------------------

Check the read depth 
Check for singletons 
```{r}

#load
load("D:\\Grad_School\\R_Projects\\BeeProject_Metagenomics\\Images\\Setup.RData")

PS_16S_BC_ON_BB

temp_16S <- meta(PS_16S_BC_ON_SF)

temp_ITS1R <-meta(PS_ITS1R_BC_ON_SF)
#Read depth analysis for original samples
Read_Depth_PS_16S_BC_ON_BB<-plot_read_distribution(PS_16S_BC_ON_BB, groups = "Province", 
                             plot.type = "histogram")+
  theme_biome_utils()+
  scale_x_continuous(trans='log10')+
  scale_fill_manual(values=c("#111111"))+ 
  geom_vline(xintercept = 500, colour = "black", linetype="dashed")+
  theme(legend.position="none")+
  labs(x = "", y = "Count")

Read_Depth_PS_16S_BC_ON_BB

PS_16S_BC_ON_BB_Pruned

PS_ITS1R_BC_ON_SF

summarize_phyloseq(PS_16S_BC_ON_SF)

#Number of singletons = 4563

#We need to remove the so we remove any taxa sums that are less than 10 reads 
PS_16S_BC_ON_BB_Pruned = prune_taxa(taxa_sums(PS_16S_BC_ON_BB) > 02, PS_16S_BC_ON_BB)

PS_16S_BC_ON_BB_Pruned

summarize_phyloseq(PS_16S_BC_ON_BB_Pruned)
```
Diversity analysis 
```{r Alpha , Shannon and Pileou Analysis}

PS_16S_BC_ON_BB_Pruned
# Alpha diversity -> microbiotaprocess    #####

alphaobj_16S_BB<- get_alphaindex(PS_16S_BC_ON_BB_Pruned)
#alphaobj2_PS_Global<-as.data.frame(alphaobj_PS_Global)
#head(alphaobj2_PS_Global)

##### violon_plots (Status) ######

Observe_status <- ggbox(alphaobj_16S_BB, 
                        geom="violin", 
                        factorNames="Province",   
                        compare = TRUE,
                        testmethod = "wilcox.test",
                        signifmap = TRUE,
                        indexNames="Observe")+ 
  theme(aspect.ratio = 0.5)+
  theme( legend.position="none")+
  theme(strip.background = element_rect(colour=NA, fill="grey"))+
  scale_fill_manual(values=c("#990038", 
                                      "#01AED9",
                                      "#999999",
                                      "#1B9E77", 
                                      "#000033", 
                                      "#FD9347"))
                                      
Shannon_status<- ggbox(alphaobj_16S_BB, 
                       geom="violin", 
                       factorNames="Province",   
                       compare = TRUE, 
                       testmethod = "wilcox.test",
                       signifmap = TRUE,
                       indexNames="Shannon")+ 
  theme(legend.position="Bottom")+
  theme(aspect.ratio = 0.5)+
  theme(strip.background = element_rect(colour=NA, fill="grey"))+
  scale_fill_manual(values=c("#990038", 
                                      "#01AED9",
                                      "#999999",
                                      "#1B9E77", 
                                      "#000033", 
                                      "#FD9347"))
                                      

Pielou_status<- ggbox(alphaobj_16S_BB, 
                      geom="violin", 
                      factorNames="Province",   
                      compare = TRUE, 
                      testmethod = "wilcox.test",
                      signifmap = TRUE,
                      indexNames="Pielou")+ 
  theme(legend.position="Bottom")+
  theme(aspect.ratio = 0.5)+
  theme(strip.background = element_rect(colour=NA, fill="grey"))+
  scale_fill_manual(values=c("#990038", 
                                      "#01AED9",
                                      "#999999",
                                      "#1B9E77", 
                                      "#000033", 
                                      "#FD9347"))


plot_grid(Observe_status+theme(legend.position="none"),
          Shannon_status+theme(legend.position="none"),
          align="vh",
          labels = c("A", "B"),
          hjust = -1,
          vjust= 3.5,
          ncol=1)      
                                

plot_grid(Observe_status+theme(legend.position="none"),
          Shannon_status+theme(legend.position="none"),
          Pielou_status+theme(legend.position="none"),
          align="vh",
          labels = c("A", "B", 'C'),
          hjust = -1,
          vjust= 3.5,
          ncol=1)

```
Check for Ubiquitous Fungi and remove them 
```{r Removal and Reabundance - BCC, echo=FALSE, eval=TRUE, include=FALSE, cache=TRUE}

top_taxa(PS_16S_BC_ON_BB_Pruned)

classtaxa_PS_16S_BC_ON_BB_Pruned<- get_taxadf(obj=PS_16S_BC_ON_BB_Pruned, taxlevel=7)
# The 30 most abundant taxonomy will be visualized by default (parameter `topn=30`). 
TopTaxa_16S_BC_ON_BB_Pruned<- ggbartax(obj=PS_16S_BC_ON_BB_Pruned, 
                   facetNames="Province", 
                   plotgroup=TRUE, 
                   topn=50) +
  xlab(NULL) +
  ylab("Relative abundance (%)") +
  guides(fill= guide_legend(keywidth = 0.5, keyheight = 0.5, ncol=4))

TopTaxa_16S_BC_ON_BB_Pruned

PS_16S_BC_ON_SF_Pruned

### RESUNME HERE

#Take a closer look at the species we can remove by creating a taxa table
Taxa_Table_16S_BC_ON_BB_Pruned <- as.data.frame(phyloseq::tax_table(PS_16S_BC_ON_BB_Pruned))
Taxa_Table_16S_BC_ON_BB_Pruned

# Removal of ubiqitous species and unknown phylums 
PS_16S_BC_ON_SF_Pruned

PS_16S_BC_ON_BB_Filtered<-subset_taxa(PS_16S_BC_ON_BB_Pruned, Phylum!="Cyanobacteria")
PS_16S_BC_ON_BB_Filtered<-subset_taxa(PS_16S_BC_ON_BB_Filtered, Phylum!="NA")
PS_16S_BC_ON_BB_Filtered<-subset_taxa(PS_16S_BC_ON_BB_Filtered, Order !="Chloroplast")
PS_16S_BC_ON_BB_Filtered<-subset_taxa(PS_16S_BC_ON_BB_Filtered, Species !="mitochondria")
PS_16S_BC_ON_BB_Filtered<-subset_taxa(PS_16S_BC_ON_BB_Filtered, Species !="metagenome")


### Did removal work ###

classtaxa_16S_BC_ON_BB_Filtered<- get_taxadf(obj=PS_16S_BC_ON_BB_Filtered, taxlevel=7)
# The 30 most abundant taxonomy will be visualized by default (parameter `topn=30`). 
TopTaxa_16S_BC_ON_BB_Filtered<- ggbartax(obj=classtaxa_16S_BC_ON_BB_Filtered, 
                   facetNames="Province", 
                   plotgroup=TRUE, 
                   topn=50) +
  xlab(NULL) +
  ylab("Relative abundance (%)") +
  guides(fill= guide_legend(keywidth = 0.5, keyheight = 0.5, ncol=4))

TopTaxa_16S_BC_ON_BB_Filtered

summarize_phyloseq(PS_16S_BC_ON_BB_Filtered)


#prunved vs filteredd
PS_16S_BC_ON_BB_Pruned
PS_16S_BC_ON_BB_Filtered

```

General Heat Map 

```{r Heat Map }

#General HeatMap 

Meco_16S_BB <- phyloseq2meco(PS_16S_BC_ON_BB_Filtered)

Heatmap_16S_BB <- trans_abund$new(dataset = Meco_16S_BB, 

                      taxrank = "Species", 

                      ntaxa = 11)

Heatmap_16S_BB$plot_heatmap(facet = c("Province","Plant","SampleType"),
                color_values = rev(RColorBrewer::brewer.pal(n = 11, 
                                                            name = "Spectral")),
                xtext_keep = FALSE, 
                withmargin = TRUE)

#BC and ON stone fruit separated metadata
PS_BC_Only_SF <- subset_samples(PS_BC_ON_SF, Province == "BC")

meta(PS_BC_Only_SF)
#82 Rows for the BC subset 

PS_ON_Only_SF <- subset_samples(PS_BC_ON_SF, Province == "ON")

meta(PS_ON_Only_SF)
#56 Rows

Taxa_Table_16S_BC_ON_SF_Pruned

```

