---
title: "Setting up new Phyloseq Objects"
author: "Minuka Hewapathirana"
coauthor:  "Herve Van de Hayden"
date: "2024-07-19"
output: github_document
---
### (1) Loading Packages 

A comprehensive list of packages required to perform metagenomics analysis for *ION torrent* data
```{r Setup, echo=TRUE, results='asis', warning= FALSE, message= FALSE}
library("knitr")
library("qiime2R") # devtools::install_github("jbisanz/qiime2R")
library("phyloseq")
library("readxl")      
library("tibble")
library("vegan")
library("DESeq2") #BiocManager::install("DESeq2")
library("speedyseq") # remotes::install_github("mikemc/speedyseq") 
library("ape")
library("ggstar")
library("forcats")
library("patchwork")
library("ggpubr")
library("plotROC")
library("viridis")
library("cowplot")
library("ggplot2")
library("microbiome") # BiocManager::install("microbiome")
library("microbiomeutilities")

library("ggtree") # BiocManager::install("ggtree")
library("ggtreeExtra") #install.packages("ggExtra")
library('MicrobiotaProcess') # BiocManager::install("MicrobiotaProcess")
#library("tidytree")
library("file2meco")
library("microeco")

```

### (2) Converting Qiime Artifacts to Phyloseq Objects

Each **Phyloseq** object requires 
- features (table.qza)
- tree (rooted-tree.qza)
- taxonomy (taxonomy.qza)
- metadata ("sample.tsv") : The metadata file shoudl be in .tsv format

```{r Global Object Creation, echo=TRUE, results='asis', warning= FALSE, message= FALSE}

#Creating a phyloseq object for the ITS1 Reverse Reads 
PS_ITS1R_Global<-qza_to_phyloseq(
  features="D:\\Grad_School\\Grad_Project\\Qiime_Outputs\\ITS2_NEW\\table.qza",
  tree="D:\\Grad_School\\Grad_Project\\Qiime_Outputs\\ITS2_NEW\\rooted-tree.qza",
  taxonomy="D:\\Grad_School\\Grad_Project\\Qiime_Outputs\\ITS2_NEW\\taxonomy.qza",
  metadata = "D:\\Grad_School\\Grad_Project\\Qiime_Outputs\\ITS2_NEW\\ITS2_Global.tsv"
)

#Checking if objects exist
PS_ITS1R_Global

# Creating a phyloseq object for the 16S Reverse Reads 
PS_16S_Global<-qza_to_phyloseq(
  features="D:\\Grad_School\\Grad_Project\\Qiime_Outputs\\New_Silva_Global\\table.qza",
  tree="D:\\Grad_School\\Grad_Project\\Qiime_Outputs\\New_Silva_Global\\rooted-tree.qza",
  taxonomy="D:\\Grad_School\\Grad_Project\\Qiime_Outputs\\New_Silva_Global\\taxonomy.qza",
  metadata = "D:\\Grad_School\\Grad_Project\\Qiime_Outputs\\New_Silva_Global\\Global.tsv"
)

PS_16S_Global

```

### (3) Obtaining StoneFruit and Blueberry ITS1 Datasets

Goal : Split ITS1 PS Object into two smaller PS objects based on StoneFruit and Blueberry in BC and ON 

- First remove Alberta from ITS1R and 16S objects

- Then split based on Stonefruits and Blueberries respectively 
```{r Splitting ITS1R Objects, echo=TRUE, results='asis', warning= FALSE, message= FALSE}

#removal of AB province
PS_ITS1R_BC_ON <- subset_samples(PS_ITS1R_Global, Province !="AB")

#removal of Wild forage 
PS_ITS1R_BC_ON_SF <- subset_samples(PS_ITS1R_BC_ON, Plant !="Wild forage")
#removal of Blueberry 
PS_ITS1R_BC_ON_SF <- subset_samples(PS_ITS1R_BC_ON_SF, Plant !="Blueberry")
#PS Object with only Stonefruit in BC and ON 
PS_ITS1R_BC_ON_SF
#Check metadata file for Stonefruit
metadata_PS_ITS1R_BC_ON_SF <- meta(PS_ITS1R_BC_ON_SF)

kable(head(metadata_PS_ITS1R_BC_ON_SF), digits = 2, align = c(rep("l", 4), rep("c", 4), rep("r", 4)))

#Remove other plant types to obtain Wild Forage and Bluberry subset 
#Cherry, Apple , Peach , Apricot 
PS_ITS1R_BC_ON_BB <-subset_samples(PS_ITS1R_BC_ON, Plant !="Cherry")
PS_ITS1R_BC_ON_BB <-subset_samples(PS_ITS1R_BC_ON_BB, Plant !="Apple")
PS_ITS1R_BC_ON_BB <- subset_samples(PS_ITS1R_BC_ON_BB, Plant !="Peach")
PS_ITS1R_BC_ON_BB <- subset_samples(PS_ITS1R_BC_ON_BB, Plant !="Apricot")

PS_ITS1R_BC_ON_BB

#Check metadatafile for blueberry
metadata_ITS1R_BC_ON_BB <- meta(PS_ITS1R_BC_ON_BB)

kable(head(metadata_ITS1R_BC_ON_BB), digits = 2, align = c(rep("l", 4), rep("c", 4), rep("r", 4)))
```

### (4) Obtaining StoneFruit and Blueberry 16S Datasets

Goal : Repeat Step 3 for 16S 

```{r Splitting 16S Objects, echo=TRUE, results='asis', warning= FALSE, message= FALSE}
#removal of AB province
PS_16S_BC_ON <- subset_samples(PS_16S_Global, Province !="AB")

#meta(PS_16S_BC_ON)

#Stone fruit subset     
PS_16S_BC_ON_SF <- subset_samples(PS_16S_BC_ON, Plant !="Wild forage")
PS_16S_BC_ON_SF <- subset_samples(PS_16S_BC_ON_SF, Plant !="Blueberry")
PS_16S_BC_ON_SF <- subset_samples(PS_16S_BC_ON_SF, Plant !="Kit")

PS_16S_BC_ON_SF

#meta(PS_16S_BC_ON_SF)

#Remove these to obtain Wild Forage and Bluberry subset 
# Cherry, Apple , Peach , Apricot 
PS_16S_BC_ON_BB <-subset_samples(PS_16S_BC_ON, Plant !="Cherry")
PS_16S_BC_ON_BB <-subset_samples(PS_16S_BC_ON_BB, Plant !="Apple")
PS_16S_BC_ON_BB <- subset_samples(PS_16S_BC_ON_BB, Plant !="Peach")
PS_16S_BC_ON_BB <- subset_samples(PS_16S_BC_ON_BB, Plant !="Apricot")
PS_16S_BC_ON_BB <- subset_samples(PS_16S_BC_ON_BB, Plant !="Kit")

PS_16S_BC_ON_BB
#meta(PS_16S_BC_ON_BB)
```

We have obtained **4 PS objects** that will be used in the following chapters to perorm metagenomics based visualizations and develop insights

### *UP NEXT* - Chapter 2 : Metagenomics Analysis of Fungal (ITS1) and bacterial (16S) data to investigate agricultural and invasive pathogen detections in **Stonefruit Systems**

